### PR TITLE
Huggingface Dataclass

### PR DESCRIPTION
**Pull request by AvindaShamal**
Huggingface dataclass and its relavent example code were removed from the main branch and added to a new branch called Huggingface-feature.

### CODE REVIEW

File: avinda_shamal/test/unit/test_hf_get_item.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            {"img": torch.rand(3, 44, 44), "label": "dog"},
+            {"img": torch.rand(3, 96, 96), "label": "cat"},
+            {"img": torch.rand(3, 100, 100), "label": "ship"},
+            {"img": torch.rand(3, 80, 80), "label": "truck"},
+        ],
+        "test": [
+            {"img": torch.rand(3, 36, 36), "label": "airplane"},
+            {"img": torch.rand(3, 58, 58), "label": "deer"},
+        ],
+    }
```
Comment (OshanJayawardena) [inline avinda_shamal/test/unit/test_hf_get_item.py]: input dataset type is incorrect, it should one from the two types below

- from datasets import Dataset, DatasetDict

File: avinda_shamal/test/unit/test_hf_len.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                "test": [
+                    {"img": "image4", "label": "label4"},
+                    {"img": "image5", "label": "label5"},
+                ],
+            },
+            "test",
+            2,
+        ),
+    ],
+)
```
Comment (OshanJayawardena) [inline avinda_shamal/test/unit/test_hf_len.py]: input dataset formats are incorrect

**Review by OshanJayawardena — CHANGES_REQUESTED** [review]
Please fix the input dataset formats

File: avinda_shamal/test/unit/test_hf_get_item.py (hunk 1 above)
Comment (AvindaShamal) [reply avinda_shamal/test/unit/test_hf_get_item.py]: done

File: avinda_shamal/test/unit/test_hf_len.py (hunk 2 above)
Comment (AvindaShamal) [reply avinda_shamal/test/unit/test_hf_len.py]: done

**Review by OshanJayawardena — APPROVED** [review]
one more request

**Review by OshanJayawardena — CHANGES_REQUESTED** [review]
Please include all tests for the data class in a single test file

**Review by OshanJayawardena — APPROVED** [review]
good to go

**Pull request merged**